src/middleware.ts

```diff
@@ -66,8 +70,41 @@ const {
   API_URL,
   SEGMENT_WRITE_KEY_SERVER,
   SENTRY_PRIVATE_DSN,
+  MEMORY_PAGE_URL_FILTER,
 } = config
 
+// For full page cache testing, find all the modern routes and enable pages we
+// would like to test.
+let cachableModernRoutes: string[] = []
+const routes = getAppRoutes()
+const appRoutes = routes[0]
+if (appRoutes) {
+  appRoutes.children?.forEach(route => {
+    const childRoutes =
+      route.children
+        ?.map(child => child.path)
+        .filter(route => route !== "/" && route !== "*")
+        .map(childPath => route.path + "/" + childPath) || []
+    cachableModernRoutes = cachableModernRoutes
+      .concat(childRoutes)
+      .concat(route.path || [])
+  })
+
+  if (MEMORY_PAGE_URL_FILTER && MEMORY_PAGE_URL_FILTER.split(",").length > 0) {
+    const cacheFilters = MEMORY_PAGE_URL_FILTER.split(",")
+    cachableModernRoutes = cachableModernRoutes.filter(route => {
+      for (const cacheFilter of cacheFilters) {
+        if (route.startsWith(cacheFilter)) {
+          return true
+        }
+      }
+      return false
+    })
+  }
+} else {
+  cachableModernRoutes = []
+}
+
```

Comment:
Can we move this chunk of code into its own file and import it. (No need to have include logic where the rest of the file is a series of middleware declarations.)